### PR TITLE
simplification suggestion

### PR DESCRIPTION
**Pull request by honeyankit**
Suggestions provided by @jakecoffman to simplify few code. Thank you!

### CODE REVIEW

**Review by jakecoffman — APPROVED** [review]
😏

File: docker/lib/dependabot/docker/utils/credentials_finder.rb [hunk 1]
```diff
@@ -27,24 +26,15 @@ def credentials_for_registry(registry_hostname)
...
-          false
-        end
-
-        def fetch_base_registry
-          return @registry_details["registry"] if @registry_details
+        def base_registry
+          @base_registry ||= credentials.find do |cred|
+            cred["type"] == "docker_registry" && cred["replaces-base"] == true
+          end
+          @base_registry ||= { "registry" => DEFAULT_DOCKER_HUB_REGISTRY, "credentials" => nil }
```
Comment (honeyankit) [inline docker/lib/dependabot/docker/utils/credentials_finder.rb:33]: @jakecoffman DEFAULT_DOCKER_HUB_REGISTRY is initialization is missing in credentials_finder.rb. It was used in two places in file_pasrser.rb. Can we make this common to both files

Comment (jakecoffman) [reply docker/lib/dependabot/docker/utils/credentials_finder.rb:33]: Ah I missed that. That might be an indicator we could push more down into the credentials_finder.

**Comment by honeyankit** [comment]
Need to redo the logic of `return credentials_finder.fetch_base_registry if credentials_finder.replaces_base?` in file_parser.rb since, we have now removed `replaces_base?` function in credentials_finder.rb

**Comment by jakecoffman** [comment]
@honeyankit Sorry I didn't actually run the tests the first time, this should be working now.

**Comment by honeyankit** [comment]
> @honeyankit Sorry I didn't actually run the tests the first time, this should be working now.

No worries. Thanks for simplying the code.

File: docker/lib/dependabot/docker/update_checker.rb [hunk 2]
```diff
@@ -318,7 +318,7 @@ def numeric_version_from(tag)
 
       def registry_hostname
         return dependency.requirements.first[:source][:registry] if dependency.requirements.first[:source][:registry]
-        return credentials_finder.fetch_base_registry if credentials_finder.replaces_base?
+        return credentials_finder.base_registry if credentials_finder.base_registry
 
         "registry.hub.docker.com"
```
Comment (jakecoffman) [inline docker/lib/dependabot/docker/update_checker.rb:323]: I guess this line is redundant now.

Comment (honeyankit) [reply docker/lib/dependabot/docker/update_checker.rb:323]: You mean the line 323?

Comment (jakecoffman) [reply docker/lib/dependabot/docker/update_checker.rb]: Yep, since the base_registry returns registry.hub.docker.com.

**Comment by jakecoffman** [comment]
It looks like there's more duplication with `using_dockerhub?` in update_checker, but that can be saved for another day I suppose.

File: docker/lib/dependabot/docker/update_checker.rb (hunk 2 above)
Comment (honeyankit) [reply docker/lib/dependabot/docker/update_checker.rb]: Yes, you are right, we can remove that line.

**Pull request merged**